zokrates_core/src/ir/expression.rs

```diff
@@ -105,14 +105,27 @@ impl<T: Field> Add<LinComb<T>> for LinComb<T> {
 
     fn add(self, other: LinComb<T>) -> LinComb<T> {
         let mut res = self.0;
+        let mut to_remove = BTreeSet::new();
+
         for (k, v) in other.0 {
-            let new_val = v + res.get(&k).unwrap_or(&T::zero());
-            if new_val == T::zero() {
-                res.remove(&k)
+            let var;
```

Comment:
We try not to declare without allocating when possible. I think it's possible here?